### ISSUE TITLE
Large database hung up on writing manifest after 100% download and sketch

### ISSUE THREAD

**Issue by ccbaumler** [sourmash-bio/sourmash_plugin_directsketch#98]
# Hung up process

I am currently working on completing the GTDB and GenBank updates. This requires ~200,000 and ~1,000,000 sequences to be gathered and sketched. Typically, there has been no issue with `directsketch` specifically `gbsketch`. Currently, I have had both sets of sequences complete the download and sketch process to hang on the `Writing manifest` step.

https://github.com/dib-lab/farm-notes/issues/68

This job and node have a the hung process. I expect the last line in the log file displayed below to have `Writing manifest`. I stayed at `Starting accession ../.. (100%)` for days.

```
==========================================
SLURM_JOB_ID = 15232609
SLURM_NODELIST = bm8
==========================================
Loaded 1032189 rows (including 1032129 rows with valid URL).
Starting accession 10321/1032189 (1%)
Starting accession 20642/1032189 (2%)
Starting accession 30963/1032189 (3%)
.
.
.
Starting accession 1011458/1032189 (98%)
Starting accession 1021779/1032189 (99%)
Starting accession 1032100/1032189 (100%)
```

After finding the node (`bm8`) in the log file, I ran a `srun` to access that node.
```
srun -p bmh --pty -w bm8 --time=5:00:00 /bin/bash
```

I than ran `top -u baumlerc` to see the jobs that I am running and the cpu it is using.

```
Tasks: 1374 total,  28 running, 1346 sleeping,   0 stopped,   0 zombie
%Cpu(s): 48.3 us,  3.1 sy,  0.0 ni, 48.4 id,  0.1 wa,  0.0 hi,  0.1 si,  0.0 st
MiB Mem : 1027920.+total, 334441.6 free, 129636.7 used, 563842.0 buff/cache
MiB Swap: 1310720.+total, 1306681.+free,   4038.9 used. 892982.9 avail Mem
Maximum tasks = 0, change to (0 is unlimited)
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
1248542 baumlerc  20   0   12404   4608   3072 R   2.9   0.0   8:43.07 top
 230974 baumlerc  20   0    2896   1536   1536 S   0.0   0.0   0:00.01 slurm_script
 230976 baumlerc  20   0  165592   6104   4608 S   0.0   0.0   0:01.65 python3.11
 231022 baumlerc  20   0    7900   3072   3072 S   0.0   0.0   0:00.01 bash
 231025 baumlerc  20   0   10.9g   5.7g  26112 S   0.0   0.6   8630:53 sourmash
1085422 baumlerc  20   0   13780   9052   3072 S   0.0   0.0   0:00.13 bash
```

Since this is showing that the job `sourmash` (which is my directsketch job) is using 0% cpu. The script must be waiting on something to complete...

## More testing

```
$ srun -p bmh --mem=50gb --pty --time=5:00:00 /bin/bash
```

## 9 sequences

```bash!
head data/missing-genomes.20240904-bacteria.csv > test/sub-missing-genomes.20240904-bacteria.csv
```

```bash!
$ sourmash scripts gbsketch test/sub-missing-genomes.20240904-bacteria.csv -o test/sub-missing-genomes.20240904-bacteria.zip --failed test/sub-missing-genomes.20240904-bacteria.failed --param-str "dna,k=21,k=31,k=51,scaled=1000,abund" -r 5 -g 2> test/sub-missing-genomes.20240904-bacteria.log

Loaded 9 rows (including 9 rows with valid URL).
Starting accession 4/9 (44%)
Starting accession 1/9 (11%)
Starting accession 5/9 (56%)
Starting accession 6/9 (67%)
Starting accession 7/9 (78%)
Starting accession 8/9 (89%)
Starting accession 9/9 (100%)
Starting accession 2/9 (22%)
Starting accession 3/9 (33%)
Writing manifest
```

```
$ sourmash sig summarize test/sub-missing-genomes.20240904-bacteria.zip

== This is sourmash version 4.8.8. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

** loading from 'test/sub-missing-genomes.20240904-bacteria.zip'
path filetype: ZipFileLinearIndex
location: /group/ctbrowngrp4/2024-ccbaumler-genbank/genbank-20240904/test/sub-missing-genomes.20240904-bacteria.zip
is database? yes
has manifest? yes
num signatures: 27
** examining manifest...
total hashes: 96323
summary of sketches:
   9 sketches with DNA, k=51, scaled=1000, abund      32124 total hashes
   9 sketches with DNA, k=31, scaled=1000, abund      32264 total hashes
   9 sketches with DNA, k=21, scaled=1000, abund      31935 total hashes
```

## 1000 sequences

```bash!
head -n 1001 data/missing-genomes.20240904-bacteria.csv > test/sub1000-missing-genomes.20240904-bacteria.csv
```

```bash!
$ sourmash scripts gbsketch test/sub1000-missing-genomes.20240904-bacteria.csv -o test/sub-missing-genomes.20240904-bacteria.z
ip --failed test/sub-missing-genomes.20240904-bacteria.failed --param-str "dna,k=21,k=31,k=51,scaled=1000,abund" -r 5 -g 2> test/sub-missing-genomes.20240904-bacteria.log
Loaded 1000 rows (including 1000 rows with valid URL).
Starting accession 30/1000 (3%)
Starting accession 40/1000 (4%)
Starting accession 50/1000 (5%)
Starting accession 60/1000 (6%)
Starting accession 70/1000 (7%)
Starting accession 160/1000 (16%)
Starting accession 80/1000 (8%)
Starting accession 170/1000 (17%)
Starting accession 90/1000 (9%)
Starting accession 180/1000 (18%)
Starting accession 100/1000 (10%)
Starting accession 110/1000 (11%)
Starting accession 120/1000 (12%)
Starting accession 130/1000 (13%)
Starting accession 140/1000 (14%)
Starting accession 150/1000 (15%)
Starting accession 190/1000 (19%)
Starting accession 200/1000 (20%)
Starting accession 210/1000 (21%)
Starting accession 220/1000 (22%)
Starting accession 230/1000 (23%)
Starting accession 240/1000 (24%)
Starting accession 250/1000 (25%)
Starting accession 260/1000 (26%)
Starting accession 270/1000 (27%)
Starting accession 280/1000 (28%)
Starting accession 290/1000 (29%)
Starting accession 300/1000 (30%)
Starting accession 310/1000 (31%)
Starting accession 320/1000 (32%)
Starting accession 330/1000 (33%)
Starting accession 340/1000 (34%)
Starting accession 350/1000 (35%)
Starting accession 360/1000 (36%)
Starting accession 370/1000 (37%)
Starting accession 380/1000 (38%)
Starting accession 390/1000 (39%)
Starting accession 400/1000 (40%)
Starting accession 410/1000 (41%)
Starting accession 420/1000 (42%)
Starting accession 430/1000 (43%)
Starting accession 440/1000 (44%)
Starting accession 450/1000 (45%)
Starting accession 460/1000 (46%)
Starting accession 470/1000 (47%)
Starting accession 480/1000 (48%)
Starting accession 490/1000 (49%)
Starting accession 500/1000 (50%)
Starting accession 510/1000 (51%)
Starting accession 520/1000 (52%)
Starting accession 530/1000 (53%)
Starting accession 540/1000 (54%)
Starting accession 550/1000 (55%)
Starting accession 560/1000 (56%)
Starting accession 10/1000 (1%)
Starting accession 20/1000 (2%)
Starting accession 570/1000 (57%)
Starting accession 580/1000 (58%)
Starting accession 590/1000 (59%)
Starting accession 600/1000 (60%)
Starting accession 610/1000 (61%)
Starting accession 620/1000 (62%)
Starting accession 630/1000 (63%)
Starting accession 640/1000 (64%)
Starting accession 650/1000 (65%)
Starting accession 660/1000 (66%)
Starting accession 670/1000 (67%)
Starting accession 680/1000 (68%)
Starting accession 690/1000 (69%)
Starting accession 700/1000 (70%)
Starting accession 710/1000 (71%)
Starting accession 720/1000 (72%)
Starting accession 730/1000 (73%)
Starting accession 740/1000 (74%)
Starting accession 750/1000 (75%)
Starting accession 760/1000 (76%)
Starting accession 770/1000 (77%)
Starting accession 780/1000 (78%)
Starting accession 790/1000 (79%)
Starting accession 800/1000 (80%)
Starting accession 810/1000 (81%)
Starting accession 820/1000 (82%)
Starting accession 830/1000 (83%)
Starting accession 840/1000 (84%)
Starting accession 850/1000 (85%)
Starting accession 860/1000 (86%)
Starting accession 870/1000 (87%)
Starting accession 880/1000 (88%)
Starting accession 910/1000 (91%)
Starting accession 890/1000 (89%)
Starting accession 900/1000 (90%)
Starting accession 920/1000 (92%)
Starting accession 930/1000 (93%)
Starting accession 940/1000 (94%)
Starting accession 950/1000 (95%)
Starting accession 960/1000 (96%)
Starting accession 970/1000 (97%)
Starting accession 1000/1000 (100%)
Starting accession 980/1000 (98%)
Starting accession 990/1000 (99%)
Writing manifest
```

```
$ sourmash sig summarize test/sub-missing-genomes.20240904-bacteria.zip

== This is sourmash version 4.8.8. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

** loading from 'test/sub-missing-genomes.20240904-bacteria.zip'
path filetype: ZipFileLinearIndex
location: /group/ctbrowngrp4/2024-ccbaumler-genbank/genbank-20240904/test/sub-missing-genomes.20240904-bacteria.zip
is database? yes
has manifest? yes
num signatures: 3000
** examining manifest...
total hashes: 6587472
summary of sketches:
   1000 sketches with DNA, k=21, scaled=1000, abund   2180462 total hashes
   1000 sketches with DNA, k=31, scaled=1000, abund   2210126 total hashes
   1000 sketches with DNA, k=51, scaled=1000, abund   2196884 total hashes
```

## 10000 sequences

```
 head -n 10001 data/missing-genomes.20240904-bacteria.csv > test/sub10000-missing-genomes.20240904-bacteria.csv
```

```bash!
$ sourmash scripts gbsketch test/sub10000-missing-genomes.20240904-bacteria.csv -o test/sub-missing-genomes.20240904-bacteria.zip --failed test/sub-miss
ing-genomes.20240904-bacteria.failed --param-str "dna,k=21,k=31,k=51,scaled=1000,abund" -r 5 -g 2> test/sub-missing-genomes.20240904-bacteria.log

Loaded 10000 rows (including 10000 rows with valid URL).
Starting accession 100/10000 (1%)
Starting accession 200/10000 (2%)
Starting accession 300/10000 (3%)
Starting accession 400/10000 (4%)
Starting accession 500/10000 (5%)
Starting accession 600/10000 (6%)
Starting accession 700/10000 (7%)
Starting accession 800/10000 (8%)
Starting accession 900/10000 (9%)
Starting accession 1000/10000 (10%)
Starting accession 1100/10000 (11%)
Starting accession 1200/10000 (12%)
Starting accession 1300/10000 (13%)
Starting accession 1400/10000 (14%)
Starting accession 1500/10000 (15%)
Starting accession 1600/10000 (16%)
Starting accession 1700/10000 (17%)
Starting accession 1800/10000 (18%)
Starting accession 1900/10000 (19%)
Starting accession 2000/10000 (20%)
Starting accession 2100/10000 (21%)
Starting accession 2200/10000 (22%)
Starting accession 2300/10000 (23%)
Starting accession 2400/10000 (24%)
Starting accession 2500/10000 (25%)
Starting accession 2600/10000 (26%)
Starting accession 2700/10000 (27%)
Starting accession 2800/10000 (28%)
Starting accession 2900/10000 (29%)
Starting accession 3000/10000 (30%)
Starting accession 3100/10000 (31%)
Starting accession 3200/10000 (32%)
Starting accession 3300/10000 (33%)
Starting accession 3400/10000 (34%)
Starting accession 3500/10000 (35%)
Starting accession 3600/10000 (36%)
Starting accession 3700/10000 (37%)
Starting accession 3800/10000 (38%)
Starting accession 3900/10000 (39%)
Starting accession 4000/10000 (40%)
Starting accession 4100/10000 (41%)
Starting accession 4200/10000 (42%)
Starting accession 4300/10000 (43%)
Starting accession 4400/10000 (44%)
Starting accession 4500/10000 (45%)
Starting accession 4600/10000 (46%)
Starting accession 4700/10000 (47%)
Starting accession 4800/10000 (48%)
Starting accession 4900/10000 (49%)
Starting accession 5000/10000 (50%)
Starting accession 5100/10000 (51%)
Starting accession 5200/10000 (52%)
Starting accession 5300/10000 (53%)
Starting accession 5400/10000 (54%)
Starting accession 5500/10000 (55%)
Starting accession 5600/10000 (56%)
Starting accession 5700/10000 (57%)
Starting accession 5800/10000 (58%)
Starting accession 5900/10000 (59%)
Starting accession 6000/10000 (60%)
Starting accession 6100/10000 (61%)
Starting accession 6200/10000 (62%)
Starting accession 6300/10000 (63%)
Starting accession 6400/10000 (64%)
Starting accession 6500/10000 (65%)
Starting accession 6600/10000 (66%)
Starting accession 6700/10000 (67%)
Starting accession 6800/10000 (68%)
Starting accession 6900/10000 (69%)
Starting accession 7000/10000 (70%)
Starting accession 7100/10000 (71%)
Starting accession 7200/10000 (72%)
Starting accession 7300/10000 (73%)
Starting accession 7400/10000 (74%)
Starting accession 7500/10000 (75%)
Starting accession 7600/10000 (76%)
Starting accession 7700/10000 (77%)
Starting accession 7800/10000 (78%)
Starting accession 7900/10000 (79%)
Starting accession 8000/10000 (80%)
Starting accession 8100/10000 (81%)
Starting accession 8200/10000 (82%)
Starting accession 8300/10000 (83%)
Starting accession 8400/10000 (84%)
Starting accession 8500/10000 (85%)
Starting accession 8600/10000 (86%)
Starting accession 8700/10000 (87%)
Starting accession 8800/10000 (88%)
Starting accession 8900/10000 (89%)
Starting accession 9000/10000 (90%)
Starting accession 9100/10000 (91%)
Starting accession 9200/10000 (92%)
Starting accession 9300/10000 (93%)
Starting accession 9400/10000 (94%)
Starting accession 9500/10000 (95%)
Starting accession 9600/10000 (96%)
Starting accession 9700/10000 (97%)
Starting accession 9800/10000 (98%)
Starting accession 9900/10000 (99%)
Starting accession 10000/10000 (100%)
Writing manifest
```

```
$ sourmash sig summarize test/sub-missing-genomes.20240904-bacteria.zip

== This is sourmash version 4.8.8. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

** loading from 'test/sub-missing-genomes.20240904-bacteria.zip'
path filetype: ZipFileLinearIndex
location: /group/ctbrowngrp4/2024-ccbaumler-genbank/genbank-20240904/test/sub-missing-genomes.20240904-bacteria.zip
is database? yes
has manifest? yes
num signatures: 30000
** examining manifest...
total hashes: 53794901
summary of sketches:
   10000 sketches with DNA, k=31, scaled=1000, abund  17960205 total hashes
   10000 sketches with DNA, k=21, scaled=1000, abund  17597997 total hashes
   10000 sketches with DNA, k=51, scaled=1000, abund  18236699 total hashes
```

**Comment by bluegenes** [comment]
I'm not sure what is causing this issue, but it seems like it would be helped by writing smaller, batched zipfiles, so we never have quite as large a zip to write. #102 will enable this behavior.

**Comment by bluegenes** [comment]
writing batched zipfiles now available in v0.4.0!
